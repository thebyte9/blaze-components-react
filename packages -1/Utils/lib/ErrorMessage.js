"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var ErrorMessage = function (_a) {
    var message = _a.message, icon = _a.icon;
    return (react_1["default"].createElement("div", { className: "validation", "data-testid": "validation-message" },
        react_1["default"].createElement("i", { className: "material-icons" }, icon),
        message));
};
ErrorMessage.defaultProps = {
    icon: "warning",
    message: ""
};
exports["default"] = ErrorMessage;
