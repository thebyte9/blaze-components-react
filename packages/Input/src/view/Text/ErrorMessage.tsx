import React from 'react';

export interface IErrorMessageProps {
  message: string;
  classes: string;
}

export const ErrorMessage = ({ classes, message }: IErrorMessageProps): JSX.Element => {
  return <span className={classes}>{message}</span>;
};
