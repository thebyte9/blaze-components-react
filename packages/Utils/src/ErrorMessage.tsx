import React, { FunctionComponent } from "react";

interface IErrorMessage {
  message: string | JSX.Element;
  icon?: string;
}

const ErrorMessage: FunctionComponent<IErrorMessage> = ({
  message,
  icon
}): JSX.Element => {
  return (
    <div className="validation" data-testid="validation-message">
      <i className="material-icons">{icon}</i>
      {message}
    </div>
  );
};

ErrorMessage.defaultProps = {
  icon: "warning",
  message: ""
};

export default ErrorMessage;
