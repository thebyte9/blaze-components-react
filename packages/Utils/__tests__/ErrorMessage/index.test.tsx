import '@testing-library/jest-dom';

import { fireEvent, render, screen } from '@testing-library/react';

import ErrorMessage from '../../src/ErrorMessage';
import React from 'react';

describe('ErrorMessage function', () => {
  const defaultProps = {
    icon: 'testIcon',
    message: 'testMessage',
  };

  it('should match snapshot', () => {
    const { asFragment } = render(<ErrorMessage message={<p>Text</p>} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
