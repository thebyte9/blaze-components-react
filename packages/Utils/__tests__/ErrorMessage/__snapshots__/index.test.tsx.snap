// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage function should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="validation"
    data-testid="validation-message"
  >
    <i
      class="material-icons"
    >
      warning
    </i>
    <p>
      Text
    </p>
  </div>
</DocumentFragment>
`;
